apps/zotonic_mod_authentication/src/models/m_authentication.erl

```diff
@@ -389,7 +389,7 @@ decode_token(Token, Context) ->
         {Type, UserId, Timestamp} = decode_payload(Payload),
         SiteSecret = site_auth_key(Context),
         UserSecret = user_auth_key(UserId, Context),
-        HashCheck = z_utils:hmac(sha256, <<SiteSecret/binary, UserSecret/binary>>, Payload),
+        HashCheck = crypto:mac(hmac, sha256, <<SiteSecret/binary, UserSecret/binary>>, Payload),
```

Comment:
Should we use a macro for `sha256` here too? Or maybe the use of the macro in `z_utils` is not needed; I am wondering if it is clear enough to just type `sha256` everywhere and if we want to upgrade it, it makes for an easy search and replace (while improving readability between upgrades).

Reply:
On second thought, it might be nice to have a z_crypto module or similar, where constants like these are listed - it would offer a nice overview on which algorithms are used for the respective uses. (sha256 for pickling, sha for 2fa for instance)